lib/rubygems/gemcutter_utilities.rb

```diff
@@ -119,6 +119,40 @@ def sign_in sign_in_host = nil
     end
   end
 
+  ##
+  # Fetch user's multifactor authentication settings and return if an extra
+  # OTP code is needed.
+
+  def need_otp?
+    return false if options[:suppress_mfa]
+    unless instance_variable_defined? :@mfa_level
+      response = rubygems_api_request(:get, 'api/v1/multifactor_auth') do |request|
+        request.add_field 'Authorization', api_key
+      end
+
+      # For compatibility to Gemcutters without mfa support
+      @mfa_level = case response
+                   when Net::HTTPNotFound
+                     'no_mfa'
+                   else
+                     with_response(response) { |resp| resp.body }
+                   end
+    end
+
+    @mfa_level == 'mfa_login_and_write'
```

Comment:
If there isn't a many different values for `@mfa_level`, shouldn't symbols be
favored over strings here?

Reply:
No idea, I didn't even thought about performance when I commented to be honest.
It was more about "semantics", that it seemed used as an "identifier" more than
a string.
But on the other hand, as you said this data is received as string already, if
the spec pass in theory we shouldn't need more code :-)
Anyway it was a small detail I just asked about in case it would help, feel
free to ignore.